include/matx_set.h

```diff
@@ -81,14 +79,15 @@ class set : public BaseOp<set<T, RANK, Desc, Op>> {
    * @param op
    *   Input operator
    */
-  inline set(tensor_impl_t<T, RANK, Desc> &out, const Op op) : out_(out), op_(op)
+  inline set(T &out, const Op op) : out_(out), op_(op)
   {
+    static_assert(is_matx_op_lhs<T>() == true, "Invalid operator on LHS of set/operator=");
```

Comment:
should this be named lvalue instead of lhs?

Reply:
changed